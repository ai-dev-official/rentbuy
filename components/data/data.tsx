// data/cribs.ts

const ShuttleIcon = require('react-native-vector-icons/FontAwesome').default;
const CabinIcon = require('react-native-vector-icons/MaterialIcons').default;
const FanIcon = require('react-native-vector-icons/FontAwesome5').default;
const WaterIcon = require('react-native-vector-icons/MaterialCommunityIcons').default;
const TinyIcon = require('react-native-vector-icons/MaterialIcons').default;
const LakeIcon = require('react-native-vector-icons/MaterialCommunityIcons').default;
const MansionIcon = require('react-native-vector-icons/FontAwesome6').default;
const LightIcon = require('react-native-vector-icons/MaterialCommunityIcons').default;
const TreeIcon = require('react-native-vector-icons/Foundation').default;
const CastleIcon = require('react-native-vector-icons/MaterialIcons').default;
const PlayIcon = require('react-native-vector-icons/MaterialCommunityIcons').default;

export interface Crib {
    id: number;
    icon: React.ComponentType<{ name: string; size?: number; color: string; style?: object }>;
    title: string;
    name: string;
    size: number;
}

export const cribs: Crib[] = [
    {
        id: 0,
        icon: ShuttleIcon,
        title: "space-shuttle",
        name: "Iconic",
        size:24,
    },
    {
        id: 2,
        icon: CabinIcon,
        title: "cabin",
        name: "Cabins",
        size:24,
    },
    {
        id: 3,
        icon: FanIcon,
        title: "fan",
        name: "OMG!",
        size:24,
    },
    {
        id: 4,
        icon: WaterIcon,
        title: "water-polo",
        name: "Beach front",
        size:24,
    },
    {
        id: 5,
        icon: TinyIcon,
        title: "house-siding",
        name: "Tiny homes",
        size:24,
    },
    {
        id: 6,
        icon: LakeIcon,
        title: "snowflake-variant",
        name: "Lake fron",
        size:24,
    },
    {
        id: 7,
        icon: MansionIcon,
        title: "house-flood-water",
        name: "Luxury",
        size:24,
    },
    {
        id: 8,
        icon: LightIcon,
        title: "lighthouse-on",
        name: "Amazing",
        size:24,
    },
    {
        id: 9,
        icon: TreeIcon,
        title: "social-treehouse",
        name: "Tree houses",
        size:24,
    },
    {
        id: 10,
        icon: CastleIcon,
        title: "castle",
        name: "Castles",
        size:24,
    },
    {
        id: 11, // Updated to a unique ID
        icon: PlayIcon,
        title: "cards-playing-heart-outline",
        name: "Play homes",
        size:24,
    },
];
