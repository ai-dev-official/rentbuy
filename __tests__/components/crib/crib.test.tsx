import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {it} from '@jest/globals';
import { cribs, Crib } from '../../../components/data/data.tsx';
import Cribs from '../../../components/crib/crib';
import { ScrollView, Text, View } from 'react-native';




describe('Cribs Component', () => {
    // Snapshot Test
    it('renders correctly', () => {
        const tree = renderer.create(<Cribs />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Test to check if the component renders the correct number of cribs
    it('should render all cribs', () => {
        const component = renderer.create(<Cribs />);
        const instance = component.root;

        const cribItems = instance.findAll(
            (node) => node.type === View && node.props.testID === 'crib-item'
        );

        expect(cribItems.length).toBe(cribs.length);
    });

    // Test to verify that each crib displays the correct title
    it('should render all cribs', () => {
      const component = renderer.create(<Cribs />);
      const instance = component.root;

      const cribItems = instance.findAll(
          (node) => node.type === View && node.props.testID === 'crib-item'
      );

      expect(cribItems.length).toBe(cribs.length);
  });

    // Test to verify that each crib has an icon component
    it('each crib should display an icon', () => {
        const component = renderer.create(<Cribs />);
        const instance = component.root;

        cribs.forEach((crib) => {
            const icon = instance.find(
                (node) =>
                    node.props.testID === `icon-${crib.id}`
            );
            expect(icon).toBeTruthy();
        });
    });

    it('each crib should display the correct title', () => {
      const instance = renderer.create(<Cribs />).root;

      cribs.forEach((crib) => {
          const title = instance.find(
              (node) =>
                  node.type === Text && node.props.testID === `title-${crib.id}`
          );
          expect(title.props.children).toBe(crib.name);
      });
  });

    // Test to ensure the ScrollView is horizontal
    it('should have a horizontal ScrollView', () => {
      const component = renderer.create(<Cribs />);
      const scrollView = component.root.find(
          (node) => node.type === ScrollView
      );
      expect(scrollView.props.horizontal).toBe(true);
  });

    // Test if each crib has the correct dimensions
    it('crib items should have fixed width and height', () => {
      const component = renderer.create(<Cribs />);
      const instance = component.root;

      const cribItems = instance.findAll(
          (node) => node.type === View && node.props.testID === 'crib-item'
      );

      cribItems.forEach((cribItem) => {
          expect(cribItem.props.style.width).toBe(60);
          expect(cribItem.props.style.height).toBe(60);
      });
  });
});

