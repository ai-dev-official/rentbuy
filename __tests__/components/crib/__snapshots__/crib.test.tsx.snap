// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cribs Component renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexDirection": "row",
    }
  }
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    {
      "paddingVertical": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-0"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-0"
        >
          Iconic
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-2"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-2"
        >
          Cabins
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-3"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome5Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-3"
        >
          OMG!
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-4"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󱊠
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-4"
        >
          Beach front
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-5"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-5"
        >
          Tiny homes
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-6"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰼪
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-6"
        >
          Lake fron
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-7"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome6Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-7"
        >
          Luxury
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-8"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰨀
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-8"
        >
          Amazing
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-9"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "fontcustom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-9"
        >
          Tree houses
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-10"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-10"
        >
          Castles
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 60,
        }
      }
      testID="crib-item"
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          testID="icon-11"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(103, 80, 164, 1)",
                  "fontSize": 24,
                },
                undefined,
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󱢭
          </Text>
        </View>
        <Text
          style={
            {
              "fontSize": 10,
              "marginTop": 5,
            }
          }
          testID="title-11"
        >
          Play homes
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
